sim/validator.py

```diff
@@ -84,25 +86,27 @@ def performance_test(
     btb_seq_path = os.path.join(btb_seq_path, '')
     output_path = os.path.join(output_path, '')
 
-    # # Validate Input
-    if not os.path.isdir(output_path):
+    # Run simulations if path to simulated reads not provided 
+    if not simulated_reads_path:
+        simulated_reads_path = simulate(output_path, samples)     
+    elif not os.path.isdir(output_path):
         raise Exception("Output path not found")
 
     if not os.path.isdir(btb_seq_path):
         raise Exception("Pipeline code repository not found")
 
     # Output Directories
-    results_path = os.path.join(output_path + branch, '')
+    if branch:
+        results_path = os.path.join(output_path + branch, '')
+    else:
+        results_path = os.path.join(output_path + get_branch(btb_seq_path))
```

Comment:
Potential for bugs here as well.  Line 100 is used to ensure a trailing slash at the end, but Line 102 doesn't join anything and has no trailing slash. Could be a bit more succinct as well, eg
```
branch_name = branch if branch else get_branch(btb_seq_path)
results_path = os.path.join(output_path, branch_name, '')
```

Reply:
good spot!